src/parser/core/modules/Entities.js

```diff
@@ -220,6 +119,40 @@ export default class Entities extends Module {
 		this.triggerChangeBuffStack(buff, event.timestamp, buff.stacks, 0)
 	}
 
+	refreshBuff(event, isDebuff) {
+		const entity = this.getBuffEventEntity(event)
+		if (!entity) {
+			return
+		}
+
+		let buff = entity.buffs.find(buff => buff.ability.guid === event.ability.guid && buff.end == null)
+
+		// If there's no existing buff, fake one from the start of the fight
+		if (!buff) {
+			buff = this.synthesizeBuff(event, isDebuff)
+			entity.buffs.push(buff)
+		}
+
+		// Update the buff's last refreshed time
+		const oldStacks = buff.stacks
+		buff.lastRefreshed = event.timestamp
+		buff.stacks = event.stack || 1
```

Comment:
Hm - the event type defs have `stack` as always-defined - does that need to be updated to be optional?

If it is optional, this should probably use `??` to avoid possible kludge of `0` stack values.

Reply:
It looks like the current behavior in Entities.js is that an `apply(de)buff` event applies 1 buff stack, and a `apply(de)buffstack` event is applying event.stack || 1 stacks (in the updateBuffStack method).  So I made `refresh(de)buff` behave like the updateBuffStack method, since I wasn't sure which way made more sense.  

Looking at the output from FFLogs, it looks like `refreshbuff` events are shaped more like `applybuff` events, and the stack property doesn't exist on them.  Will change this to always be 1.